datastore.go

```diff
@@ -11,11 +11,19 @@ import (
 	options "github.com/dgraph-io/badger/options"
 	ds "github.com/ipfs/go-datastore"
 	dsq "github.com/ipfs/go-datastore/query"
-	logger "github.com/ipfs/go-log"
+	logger "github.com/ipfs/go-log/v2"
 	goprocess "github.com/jbenet/goprocess"
 )
 
-var log = logger.Logger("badger")
+type badgerLog struct {
```

Comment:
Let's add a comment explaining why we're doing this.

Also, we might as well _only_ wrap the logger when opening the datastore.